src/main/kotlin/build/buf/intellij/module/ModuleDigest.kt

```diff
@@ -0,0 +1,42 @@
+// Copyright 2022-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.buf.intellij.module
+
+/**
+ * [ModuleDigest] is equivalent to `bufmodule.Digest` in the Buf CLI.
+ */
+data class ModuleDigest(val digestType: ModuleDigestType, val hex: String) {
+    private val casDigest: build.buf.intellij.cas.CASDigest =
+        build.buf.intellij.cas.CASDigest(build.buf.intellij.cas.CASDigestType.SHAKE_256, hex)
```

Comment:
why not import `build.buf.intellij.cas.CASDigest`?

Reply:
Also, does the "B5" module digest also use SHAKE256? If so, what is the difference between the two module digest types?

Reply:
> why not import `build.buf.intellij.cas.CASDigest`?

Previously I had both classes called `Digest` in different packages but ended up prefixing them so can import this now.

> Also, does the "B5" module digest also use SHAKE256? If so, what is the difference between the two module digest types?

Both "B4" and "B5" use the same hash but different algorithms for determining the digest for a module. For example, B4 includes the `buf.yaml` and `buf.lock` files (if present), while B5 instead just uses a hash of it's direct dependency hashes.